applications/IgaApplication/custom_conditions/base_discrete_condition.h

```diff
@@ -0,0 +1,319 @@
+    ) override;
+
+
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "\"BaseDiscreteCondition\" #" << Id();
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "\"BaseDiscreteCondition\" #" << Id();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const {
+        pGetGeometry()->PrintData(rOStream);
+    }
+    ///@}
+
+protected:
+
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+    * This functions calculates both the RHS and the LHS
+    * @param rLeftHandSideMatrix: The LHS
+    * @param rRightHandSideVector: The RHS
+    * @param rCurrentProcessInfo: The current process info instance
+    * @param CalculateStiffnessMatrixFlag: The flag to set if compute the LHS
+    * @param CalculateResidualVectorFlag: The flag to set if compute the RHS
+    */
+    virtual void CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag);
+
+    /**
+    * CalculateJacobian computes the mapping from Geometry to Parameter Space.
+    *
+    * @param[in] DN_De derivatives of shape functions in two directions.
+    * @param[out] Jacobian calculated Jacobian. 
+    * @param[in] rWorkingSpaceDimension GeometrySpace coordinates. For surfaces default 3.
+    * @param[in] rLocalSpaceDimension ParameterSpace coordinates For surfaces default 2.
+    */
+    void CalculateJacobian(const Matrix& DN_De,
+        Matrix& Jacobian,
+        const int rWorkingSpaceDimension = 3,
+        const int rLocalSpaceDimension = 2);
+
+    /** Gets the number of dofs per node.
+    * If a derived condition has more degrees of freedom this function has to be overridden
+    *
+    * @return Number of dofs per node.
+    */
+    virtual inline std::size_t DofsPerNode() const
+    {
+        return 3;
+    }
+
+    /** Gets the number of nodes.
+    *
+    * @return Number of nodes.
+    */
+    std::size_t inline NumberOfNodes() const
+    {
+        return GetGeometry().size();
+    }
+
+    /** Gets the number of degrees of freedom.
+    *
+    * @return Number of degrees of freedom.
+    */
+    std::size_t inline NumberOfDofs() const
+    {
+        return NumberOfNodes() * DofsPerNode();
+    }
+
+    ///@}
+private:
+    ///@name Serialization
+    ///@{
+    friend class Serializer;
+
+    virtual void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, Condition);
+    }
+
+    virtual void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, Condition);
+    }
+    ///@}
```

Comment:
I think there is also some simplified print stuff in the `IgaBaseElement` which you can use.